src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js

```diff
@@ -28,131 +28,119 @@ const propTypes = {
-        this.videoChatIconWrapper = null;
-        this.menuItemData = [
-            {
-                icon: ZoomIcon,
-                text: props.translate('videoChatButtonAndMenu.zoom'),
-                onPress: () => {
-                    this.setMenuVisibility(false);
-                    Linking.openURL(CONST.NEW_ZOOM_MEETING_URL);
-                },
+    const menuItemData = [
+        {
+            icon: ZoomIcon,
+            text: props.translate('videoChatButtonAndMenu.zoom'),
+            onPress: () => {
+                setMenuVisibility(false);
+                Linking.openURL(CONST.NEW_ZOOM_MEETING_URL);
             },
-            {
-                icon: GoogleMeetIcon,
-                text: props.translate('videoChatButtonAndMenu.googleMeet'),
-                onPress: () => {
-                    this.setMenuVisibility(false);
-                    Linking.openURL(this.props.googleMeetURL);
-                },
+        },
+        {
+            icon: GoogleMeetIcon,
+            text: props.translate('videoChatButtonAndMenu.googleMeet'),
+            onPress: () => {
+                setMenuVisibility(false);
+                Linking.openURL(props.googleMeetURL);
             },
-        ];
-
-        this.state = {
-            isVideoChatMenuActive: false,
-            videoChatIconPosition: {x: 0, y: 0},
-        };
-    }
-
-    componentDidMount() {
-        this.dimensionsEventListener = Dimensions.addEventListener('change', this.measureVideoChatIconPosition);
-    }
-
-    componentWillUnmount() {
-        if (!this.dimensionsEventListener) {
-            return;
-        }
-        this.dimensionsEventListener.remove();
-    }
-
-    /**
-     * Set the state variable isVideoChatMenuActive
-     * @param {Boolean} isVideoChatMenuActive
-     */
-    setMenuVisibility(isVideoChatMenuActive) {
-        this.setState({isVideoChatMenuActive});
-    }
+        },
+    ];
 
     /**
-     * This gets called onLayout to find the cooridnates of the wrapper for the video chat button.
+     * This gets called onLayout to find the coordinates of the wrapper for the video chat button.
      */
-    measureVideoChatIconPosition() {
-        if (!this.videoChatIconWrapper) {
+    const measureVideoChatIconPosition = useCallback(() => {
+        if (!videoChatIconWrapperRef.current) {
             return;
         }
 
-        this.videoChatIconWrapper.measureInWindow((x, y) =>
-            this.setState({
-                videoChatIconPosition: {x, y},
-            }),
-        );
-    }
+        videoChatIconWrapperRef.current.measureInWindow((x, y) => {
+            setVideoChatIconPosition({x, y});
+        });
+    }, []);
+
+    const closePopover = useCallback(() => {
+        setMenuVisibility(false);
+    }, []);
+
+    useEffect(() => {
+        dimensionsEventListenerRef.current = Dimensions.addEventListener('change', measureVideoChatIconPosition);
```

Comment:
We don't need to use a ref. We can create a `dimensionsEventListener` variable inside this `useEffect`. Please refer to this example https://react.dev/reference/react/Component#migrating-a-component-with-lifecycle-methods-from-a-class-to-a-function